### ISSUE TITLE
Refactor input and configuration

### ISSUE THREAD

**Issue by alex3683** [aixigo/freemarker-loader#3]
- [ ] Make example data input instead of the template
- [ ] Add main template as configuration property
- [ ] Derive context for further template loading from main template
- [ ] As fallback, make context for template loading configurable

**Comment by alex3683** [comment]
WIP on this branch: `feature/3-refactor-structure`

**Comment by jpommerening** [comment]
@alex3683 a few things:

- We don't need `this.addDependency( templateFile )` in the `index.js`:
  The template loader supplied to the FreeMarker configuration already adds dependencies as they are requested by FreeMarker: https://github.com/aixigo/freemarker-loader/blob/feature/3-refactor-structure/lib/utils.js#L97
- I think we can drop the need for a `templateRootDirectory` option if we use the webpack (`this.options.context || process.cwd()`) root and interpret the `template` option relative to that. Or keep using `this.context` and pass in the correct path to the template, relative to `this.resourcePath`.

I'm still not sure if I like _not_ using the template as the input for the loader as it complicates the path resolution, but I agree it's a net positive for our use case where we have _one_ template and want to switch between different data modules. So, yes, let's go with the more pragmatic solution you implemented here :+1:

**Comment by jpommerening** [comment]
Okay, I made a few small amendments (we went through them on Friday) and updated the test to reflect the changes.
Final nitpick: I changed the name of the loader's input parameter to be in line with current webpack [convention](https://webpack.js.org/development/how-to-write-a-loader/#examples). ;)

And now:

## Breaking Change 🚨

Earlier versions of this loader expected the template as the resource to load:
```
freemarker-loader?data=./substitutions.json!./some-freemarker-template.html
```

This has been swapped, so that the template is now a query parameter and the data is received as the input module to be loaded:
```
freemarker-loader?template=./some-freemarker-template.html!./substitutions.json
```

Keep in mind, that with this change, the substitution data will appear to be the loaded resource to all downstream loaders. If, for example, you use `html-loader!freemarker-loader!./input-data.js`, then all the URLs in the HTML will be _relative to the input data_ instead of the template!

**Comment by jpommerening** [comment]
Implemented on `master` (`v0.3.0`)

@alex3683: Want to have a look before I close this? Otherwise, I'll release it together with #2…

**Comment by alex3683** [comment]
Fixed the test to use a fixed locale, but otherwise this looks good. 🚤 it!

**Comment by viviansteller** [comment]
Any plans to release this?

**Comment by alex3683** [comment]
Hi @viviansteller 
up to now we weren't aware of any "third-party" users, so we waited a bit to collect some stuff ;-) I don't know if @jpommerening still has some breaking stuff in the pipeline, but if everything looks fine, we'll make a release soon.

**Comment by alex3683** [comment]
@jpommerening Just had a look at the readme. We need to adjust it according to the changes.

**Comment by jpommerening** [comment]
Updated the readme. @alex3683 just published version `0.3.0` 👍
